Sources/TableViewDataSource.swift

```diff
@@ -37,10 +36,18 @@ open class TableViewDataSource: NSObject, UITableViewDataSource, UITableViewDele
         case rowsModified
     }
 
-    public var tableViewModel: MutableProperty<TableViewModel?> = MutableProperty(nil)
     public let tableView: UITableView
 
-    private var _tableViewModel: TableViewModel? { return self.tableViewModel.value }
+    public var tableViewModel: TableViewModel? {
+        willSet {
+            assert(Thread.isMainThread, "Must set \(#function) on main thread")
```

Comment:
This is the major behavior change for users.  Making you only able to set this on Main Thread.